lib/elixir/lib/keyword.ex

```diff
@@ -571,11 +571,23 @@ defmodule Keyword do
       iex> Keyword.merge([a: 1, b: 2], [a: 3, d: 4, a: 5])
       [b: 2, a: 3, d: 4, a: 5]
 
+      iex> Keyword.merge [a: 1], [2, 3]
+      ** (ArgumentError) expected a keyword-list in second argument, got: [2, 3]
+
   """
   @spec merge(t, t) :: t
   def merge(keywords1, keywords2) when is_list(keywords1) and is_list(keywords2) do
-    fun = fn {k, _v} -> not has_key?(keywords2, k) end
-    :lists.filter(fun, keywords1) ++ keywords2
+    if keyword?(keywords2) do
+      fun = fn
+        {k, _v} when is_atom(k) ->
```

Comment:
Sorry, I wasn't clear enough. I mean it'd better to use `{key, _value}` instead.